File: data-structures/tree/python/tree.py
class Node:
    def __init__(self, value):
        self.value = value
        self.children: list[Node] = []

    def insert(self, value):
        child = Node(value)
        self.children.append(child)

    def remove(self, value):
        self.children = [
            child for child in self.children if child.value != value
        ]


class Tree:
    def __init__(self):
        self.root: Node | None = None

    def create_root(self, value):
        child = Node(value)
        self.root = child

File: data-structures/tree/typescript/tree.ts
export class Node<T> {
	children: Node<T>[] = [];

	constructor(public value: T) {
		this.value = value;
	}

	insert(value: T): void {
		const child = new Node<T>(value);
		this.children.push(child);
	}

	remove(value: T): void {
		this.children = this.children.filter(child => child.value !== value);
	}
}

export class Tree<T> {
	root: Node<T> | null;

	constructor() {
		this.root = null;
	}

	createRoot(value: T): void {
		const child = new Node<T>(value);
		this.root = child;
	}
}
